samcli/local/docker/manager.py

```diff
@@ -64,13 +125,20 @@ def run(self, container, input_data=None, warm=False):
         else:
             LOG.info("Requested to skip pulling images ...\n")
 
-        if not container.is_created():
-            # Create the container first before running.
-            # Create the container in appropriate Docker network
-            container.network_id = self.docker_network_id
-            container.create()
-
-        container.start(input_data=input_data)
+        try:
+            if not container.is_created():
+                # Create the container first before running.
+                # Create the container in appropriate Docker network
+                container.network_id = self.docker_network_id
+                container.create()
+
+            container.start(input_data=input_data)
+        except docker.errors.APIError as api_error:
+            if api_error.status_code == 409:
+                raise DockerContainerException("'{}' Docker container name is already taken".format(container.name))
```

Comment:
nit: Can we replace 'taken' with 'in use' ? Conveys the intent better

Reply:
Your version aligns more closely with actual docker error message. So why not, I will do it.
![image](https://user-images.githubusercontent.com/34073648/48972483-0ead3c80-f034-11e8-80f9-0cccbbddc26f.png)

Reply:
I've removed `is_container_name_taken` method altogether. Now it does this
```python
except docker.errors.APIError as api_error:
            raise DockerContainerException("[From Docker]\n{}".format(api_error))
```

Preceding the error message with "[From Docker]", because Docker  API error messages are pretty and informative. 

If you want I can still log.debug(exc_info=True) here also to get traceback